There is a concern on NaN value

we can drop the nan row by use of dropna()

In [3]:
import pandas as pd
df=pd.read_csv('homeData.csv')
df


time  dhi_pyr  ghi_pyr_1  ghi_pyr_2  air_temperature  \
0      4/12/2019 0:01      0.0        0.0        0.0        15.598166   
1      4/12/2019 0:02      0.0        0.0        0.0        15.573500   
2      4/12/2019 0:03      0.0        0.0        0.0        15.566666   
3      4/12/2019 0:04      0.0        0.0        0.0        15.556666   
4      4/12/2019 0:05      0.0        0.0        0.0        15.532499   
...               ...      ...        ...        ...              ...   
1434  4/12/2019 23:55      0.0        0.0        0.0        16.480665   
1435  4/12/2019 23:56      0.0        0.0        0.0        16.467333   
1436  4/12/2019 23:57      0.0        0.0        0.0        16.447332   
1437  4/12/2019 23:58      0.0        0.0        0.0        16.432333   
1438  4/12/2019 23:59      0.0        0.0        0.0        16.418999   

      relative_humidity  barometric_pressure  precipitation  wind_speed  \
0             97.782997           811.291992            0.0    1.251317   
1             97.705002           811.211609            0.0    1.101517   
2             97.707001           811.216919            0.0    0.602717   
3             97.619003           811.210022            0.0    1.107400   
4             97.594002           811.212647            0.0    0.540367   
...                 ...                  ...            ...         ...   
1434          95.468002           811.291809            0.0    2.260333   
1435          95.230003           811.272644            0.0    2.513667   
1436          95.094002           811.263000            0.0    2.193667   
1437          95.198998           811.236145            0.0    2.220333   
1438          95.293999           811.243286            0.0    2.047000   

      wind_from_direction  gti_clean  gti_soil  gti_monthly  sensor_cleaning  
0              243.790543        0.0       0.0          0.0              0.0  
1              243.470978        0.0       0.0          0.0              0.0  
2              231.842773        0.0       0.0          0.0              0.0  
3              252.289948        0.0       0.0          0.0              0.0  
4              231.998215        0.0       0.0          0.0              0.0  
...                   ...        ...       ...          ...              ...  
1434           267.461243        0.0       0.0          0.0              0.0  
1435           258.307098        0.0       0.0          0.0              0.0  
1436           256.243042        0.0       0.0          0.0              0.0  
1437           257.354492        0.0       0.0          0.0              0.0  
1438           255.351746        0.0       0.0          0.0              0.0  

[1439 rows x 14 columns]

before droping the na value categories 
The dataset was 1439 rows × 14 columns

In [4]:
df.dropna(inplace=True)
df

time  dhi_pyr  ghi_pyr_1  ghi_pyr_2  air_temperature  \
0      4/12/2019 0:01      0.0        0.0        0.0        15.598166   
1      4/12/2019 0:02      0.0        0.0        0.0        15.573500   
2      4/12/2019 0:03      0.0        0.0        0.0        15.566666   
3      4/12/2019 0:04      0.0        0.0        0.0        15.556666   
4      4/12/2019 0:05      0.0        0.0        0.0        15.532499   
...               ...      ...        ...        ...              ...   
1434  4/12/2019 23:55      0.0        0.0        0.0        16.480665   
1435  4/12/2019 23:56      0.0        0.0        0.0        16.467333   
1436  4/12/2019 23:57      0.0        0.0        0.0        16.447332   
1437  4/12/2019 23:58      0.0        0.0        0.0        16.432333   
1438  4/12/2019 23:59      0.0        0.0        0.0        16.418999   

      relative_humidity  barometric_pressure  precipitation  wind_speed  \
0             97.782997           811.291992            0.0    1.251317   
1             97.705002           811.211609            0.0    1.101517   
2             97.707001           811.216919            0.0    0.602717   
3             97.619003           811.210022            0.0    1.107400   
4             97.594002           811.212647            0.0    0.540367   
...                 ...                  ...            ...         ...   
1434          95.468002           811.291809            0.0    2.260333   
1435          95.230003           811.272644            0.0    2.513667   
1436          95.094002           811.263000            0.0    2.193667   
1437          95.198998           811.236145            0.0    2.220333   
1438          95.293999           811.243286            0.0    2.047000   

      wind_from_direction  gti_clean  gti_soil  gti_monthly  sensor_cleaning  
0              243.790543        0.0       0.0          0.0              0.0  
1              243.470978        0.0       0.0          0.0              0.0  
2              231.842773        0.0       0.0          0.0              0.0  
3              252.289948        0.0       0.0          0.0              0.0  
4              231.998215        0.0       0.0          0.0              0.0  
...                   ...        ...       ...          ...              ...  
1434           267.461243        0.0       0.0          0.0              0.0  
1435           258.307098        0.0       0.0          0.0              0.0  
1436           256.243042        0.0       0.0          0.0              0.0  
1437           257.354492        0.0       0.0          0.0              0.0  
1438           255.351746        0.0       0.0          0.0              0.0  

[1437 rows x 14 columns]

now after droping the na values:
the data set is at 1437 rows × 14 columns 
 from 1439 rows × 14 columns